homeassistant/components/lyric/__init__.py

```diff
@@ -0,0 +1,197 @@
+        raise ConfigEntryNotReady from exception
+
+    async def async_update_data() -> Lyric:
+        """Fetch data from Lyric."""
+        async with async_timeout.timeout(60):
+            try:
+                await lyric.get_locations()
+            except LYRIC_EXCEPTIONS as exception:
+                raise UpdateFailed(exception) from exception
+            return lyric
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        # Name of the data. For logging purposes.
+        name="lyric_coordinator",
+        update_method=async_update_data,
+        # Polling interval. Will only be polled if there are subscribers.
+        update_interval=timedelta(seconds=120),
+    )
+
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    # Fetch initial data so we have data when entities subscribe
+    await coordinator.async_refresh()
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, component)
+                for component in PLATFORMS
+            ]
+        )
+    )
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+class LyricEntity(CoordinatorEntity):
+    """Defines a base Honeywell Lyric entity."""
+
+    def __init__(
+        self,
+        coordinator: DataUpdateCoordinator,
+        location: LyricLocation,
+        device: LyricDevice,
+        key: str,
+        name: str,
+        icon: str,
+    ) -> None:
+        """Initialize the Honeywell Lyric entity."""
+        super().__init__(coordinator)
+        self._update_thermostat = coordinator.data.update_thermostat
+        self._device = device
+        self._location = location
+        self._key = key
+        self._name = name
+        self._icon = icon
+        self._available = True
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique ID for this entity."""
+        return self._key
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def icon(self) -> str:
+        """Return the mdi icon of the entity."""
+        return self._icon
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return self.coordinator.last_update_success and self._available
+
+
+class LyricDeviceEntity(LyricEntity):
+    """Defines a Honeywell Lyric device entity."""
+
+    @property
+    def device_info(self) -> Dict[str, Any]:
+        """Return device information about this Honeywell Lyric instance."""
+        return {
+            "identifiers": {(DOMAIN, self._device.macID)},
```

Comment:
If this is a mac address, use `connections` instead of `identifiers` https://developers.home-assistant.io/docs/device_registry_index/#defining-devices